src/irr/asset/CPLYMeshFileLoader.h

```diff
@@ -125,18 +128,21 @@ class CPLYMeshFileLoader : public asset::IAssetLoader
 
     struct SContext
     {
-        core::vector<std::unique_ptr<SPLYElement>> ElementList;
-
-		std::unique_ptr<io::IReadFile> File;
-		std::unique_ptr<char> Buffer;
+        core::vector<core::smart_refctd_ptr<SPLYElement>> ElementList;
```

Comment:
unique_ptr is the only option here, as SPLYElement does not inherit from IReferenceCounted.